Engines/Wine/Engine/Object/script.js

```diff
@@ -5,333 +5,362 @@ include("utils.functions.net.download");
-* @param {map} [userData=empty]
-* @returns {String} output
-*/
-Wine.prototype.run = function (executable, args, workingDirectory, captureOutput, wait, userData) {
-    if (!args) {
-        args = [];
-    } else if (typeof args === 'string' || args instanceof String) {
-        args = [args];
+    /**
+     *
+     * @param {string} executable
+     * @param {array} [args = []]
+     * @param {boolean} [wait=false]
+     */
+    runInsidePrefix(executable, args, wait) {
+        if (!args) {
+            args = [];
+        } else if (typeof args === "string" || args instanceof String) {
+            args = [args];
+        }
+
+        if (!wait) {
+            wait = false;
+        }
+
+        return this.run(this.prefixDirectory() + "/drive_c/" + executable, args, this.prefixDirectory(), false, wait);
     }
-    if (!workingDirectory) {
-        workingDirectory = this._implementation.getContainerDirectory(this._implementation.getWorkingContainer()) + "/drive_c";
+
+    /**
+     *
+     * @param {string} executable
+     * @param {array} [args = []]
+     * @param {string} [workingDirectory = working container]
+     * @param {boolean} [captureOutput=false]
+     * @param {boolean} [wait=false]
+     * @param {map} [userData=empty]
+     * @returns {String} output
+     */
+    run(executable, args, workingDirectory, captureOutput, wait, userData) {
+        if (!args) {
+            args = [];
+        } else if (typeof args === "string" || args instanceof String) {
+            args = [args];
+        }
+        if (!workingDirectory) {
+            workingDirectory =
+                this._implementation.getContainerDirectory(this._implementation.getWorkingContainer()) + "/drive_c";
+        }
+        if (!captureOutput) {
+            captureOutput = false;
+        }
+        if (!wait) {
+            wait = false;
+        }
+        if (!userData) {
+            userData = [];
+        }
+
+        return this._implementation.run(executable, args, workingDirectory, captureOutput, wait, userData);
     }
-    if (!captureOutput) {
-        captureOutput = false;
+
+    /**
+     * uninstall application
+     * @param {string} name of the application which shall be uninstalled
+     * @returns {bool} true if an application has been uninstalled, false otherwise
+     */
+    uninstall(application) {
+        const list = this.run("uninstaller", ["--list"], this.prefixDirectory(), true, true);
+        const appEscaped = application.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
+        const re = new RegExp("(.*)\\|\\|\\|.*" + appEscaped);
+        const uuid = list.match(re);
+        if (uuid) {
+            this._implementation.getWizard().wait(tr("Please wait while {0} is uninstalled...", application));
+            this.run("uninstaller", ["--remove", uuid[1]], this.prefixDirectory(), false, true);
+
+            return true;
+        } else {
+            print(tr("Could not uninstall {0}!", application));
```

Comment:
Should we throw an `Error` here?

Reply:
No. It is valid to try to uninstall an app which does not exist. Returning `false` should be sufficient.

Reply:
Ok